src/cwts/networkanalysis/run/FileIO.java

```diff
@@ -0,0 +1,396 @@
+
+                int node1;
+                int node2;
+                try
+                {
+                    node1 = Integer.parseUnsignedInt(columns[0]);
+                    node2 = Integer.parseUnsignedInt(columns[1]);
+                }
+                catch (NumberFormatException e)
+                {
+                    throw new IOException("Node must be represented by a zero-index integer number (line " + lineNo + ").");
+                }
+                edges[0].append(node1);
+                edges[1].append(node2);
+                if (node1 >= nNodes)
+                    nNodes = node1 + 1;
+                if (node2 >= nNodes)
+                    nNodes = node2 + 1;
+
+                if (weightedEdges)
+                {
+                    double weight;
+                    try
+                    {
+                        weight = Double.parseDouble(columns[2]);
+                    }
+                    catch (NumberFormatException e)
+                    {
+                        throw new IOException("Edge weight must be a number (line " + lineNo + ").");
+                    }
+                    edgeWeights.append(weight);
+                }
+
+                line = reader.readLine();
+            }
+        }
+        catch (FileNotFoundException e)
+        {
+            System.err.println("Error while reading edge list from file: File not found.");
+            System.exit(-1);
+        }
+        catch (IOException e)
+        {
+            System.err.println("Error while reading edge list from file: " + e.getMessage());
+            System.exit(-1);
+        }
+        finally
+        {
+            if (reader != null)
+                try
+                {
+                    reader.close();
+                }
+                catch (IOException e)
+                {
+                    System.err.println("Error while reading edge list from file: " + e.getMessage());
+                    System.exit(-1);
+                }
+        }
+
+        // Create network.
+        Network network = null;
+        int[][] edges2 = new int[2][];
+        edges2[0] = edges[0].toArray();
+        edges2[1] = edges[1].toArray();
+        try
+        {
+            if (weightedEdges)
+                network = new Network(nNodes, true, edges2, edgeWeights.toArray(), sortedEdgeList, true);
+            else
+                network = new Network(nNodes, true, edges2, sortedEdgeList, true);
+        }
+        catch (IllegalArgumentException e)
+        {
+            System.err.println("Error while creating network: " + e.getMessage());
+            System.exit(-1);
+        }
+        return network;
+    }
+
+    /**
+     * Reads a clustering from a file.
+     *
+     * @param filename Filename
+     * @param nNodes   Number of nodes
+     *
+     * @return Clustering
+     */
+    public static Clustering readClustering(String filename, int nNodes)
```

Comment:
When reading a network it doesn't require to specify the number of nodes / edges. Wouldn't it make more sense to use the same approach then also here? This would mean using the `Dynamic...Array` types, but I think the functionality would then be more convenient.